nautobot/dcim/models/devices.py

```diff
@@ -1209,3 +1217,91 @@ def __str__(self):
         if self.alias:
             return self.alias
         return f"{self.platform} - {self.version}"
+
+
+#
+# Controller
+#
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "locations",
+    "statuses",
+    "webhooks",
+)
+class Controller(PrimaryModel, ConfigContextModel):
+    """Controller model."""  # TODO: (whitej6) finialize description
+
+    name = models.CharField(max_length=255, unique=True)
+    status = StatusField(blank=False, null=False)
+    description = models.CharField(max_length=200, blank=True)
+    location = models.ForeignKey(
+        to="dcim.Location",
+        on_delete=models.PROTECT,
+        related_name="controllers",
+    )
+    platform = models.ForeignKey(
```

Comment:
Should `Controller` be linked to `SoftwareVersion` same as `Device` is?

Reply:
I think that's implicit through `Controller.deployed_controller_device.software_version`?

Reply:
Isn't `Controller->Platform` field implicit as well?